deepmd/pt/train/training.py

```diff
@@ -653,6 +653,12 @@ def run(self) -> None:
             prof.start()
 
         def step(_step_id, task_key="Default") -> None:
+            if self.multi_task:
+                model_index = dp_random.choice(
+                    np.arange(self.num_model, dtype=np.int_),
+                    p=self.model_prob,
+                )
```

Comment:
I see the hack for multiprocessing is removed. Have we ensured random seeds are different in different processes?

Reply:
See also https://numpy.org/doc/stable/reference/random/parallel.html

Reply:
> Have we ensured random seeds are different in different processes?

Yes, by not setting the seed explicitly.

https://github.com/deepmodeling/deepmd-kit/blob/e695a91ca6f7a1c9c830ab1c58b7b7a05db3da23/deepmd/utils/random.py#L9
This initializes the generator with a random state.
For torch, the `seed()` method is never called.
For tensorflow it goes here
https://github.com/deepmodeling/deepmd-kit/blob/e695a91ca6f7a1c9c830ab1c58b7b7a05db3da23/deepmd/tf/entrypoints/train.py#L220-L225

Reply:
If the seed is not set explicitly, different processes may have the same seed.

Reply:
TF seed is also wrong.

Reply:
> If the seed is not set explicitly, different processes may have the same seed.

`[np.random.RandomState](https://numpy.org/doc/stable/reference/random/legacy.html#numpy.random.RandomState)` initializes seed from random generators if the seed is not provided. So I think it would be less likely for different processes having the same seed.

Reply:
@njzjz Do you think we should mandatorily set a seed?

Reply:
> @njzjz Do you think we should mandatorily set a seed?

Indeed, it's surprising that the random seed set by the user is not used, meaning the training may not be reproducible.